src/PactNet.Native/Messaging/MessageMiddleware.cs

```diff
@@ -0,0 +1,91 @@
+#if NETSTANDARD2_0_OR_GREATER
+
+using System.IO;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+
+using Microsoft.AspNetCore.Http;
+
+using Newtonsoft.Json;
+
+namespace PactNet.Native.Messaging
+{
+    /// <summary>
+    /// Defines the message middleware
+    /// It handles the queries for the verifier when there are message interactions
+    /// </summary>
+    public class MessageMiddleware
+    {
+
+        private readonly RequestDelegate _next;
+
+        /// <summary>
+        /// Creates an instance of <see cref="MessageMiddleware"/>
+        /// </summary>
+        /// <param name="next"></param>
+        public MessageMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        /// <summary>
+        /// The invoke method of the middleware
+        /// </summary>
+        /// <param name="context">The http context</param>
+        /// <returns>A task</returns>
+        public async Task InvokeAsync(HttpContext context)
+        {
+            if (!(context.Request.Path.Value?.StartsWith(Constants.PactMessagesPath) ?? false))
+            {
+                await _next(context);
+                return;
+            }
+
+            context.Response.StatusCode = (int)HttpStatusCode.OK;
+
+            string jsonRequestBody = await GetRequestBodyAsync(context);
+
+            var interactionDescription = JsonConvert.DeserializeObject<MessageInteraction>(jsonRequestBody)?.Description;
```

Comment:
This uses Json.Net directly, but the default jSON serializer in ASP.Net Core now is `System.Text.Json`. We should probably reuse that just in case there are compatibility issues, and also then we can reuse the default options defined by the user when they create their API.

Reply:
Not sure, it's the case. Microsoft is going to package newtonsoft in their future framework because system.text hasn't caught up in time... That said, I don't see why this middleware shouldn't work with System.Text.Json.